### PR TITLE
[qt6] Use QTransform instead of QMatrix

### PR DESCRIPTION
**Pull request by m-kuhn**
QMatrix is dead

I don't know what I'm doing and completely bet on unit tests to prove me wrong

### CODE REVIEW

File: src/core/symbology/qgsellipsesymbollayer.cpp [hunk 1]
```diff
@@ -272,11 +273,11 @@ void QgsEllipseSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderContext &
     return;
   }
 
-  QMatrix transform;
+  QMatrix4x4 transform;
```
Comment (nyalldawson) [inline src/core/symbology/qgsellipsesymbollayer.cpp:276]: Couldn't this just be a QTransform?

Comment (m-kuhn) [reply src/core/symbology/qgsellipsesymbollayer.cpp:276]: Let me quote myself:

> I don't know what I'm doing and completely bet on unit tests to prove me wrong

so ... possibly :stuck_out_tongue_closed_eyes:

Comment (m-kuhn) [reply src/core/symbology/qgsellipsesymbollayer.cpp:276]: Let me quote myself:

> I don't know what I'm doing and completely bet on unit tests to prove me wrong

so ... possibly :stuck_out_tongue_closed_eyes:

Comment (nyalldawson) [reply src/core/symbology/qgsellipsesymbollayer.cpp:276]: I'm 90% sure you could just change the variable to QTransform and it'll work correctly

Comment (m-kuhn) [reply src/core/symbology/qgsellipsesymbollayer.cpp:276]: Makes a lot of sense, patch inbound

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**